File: New_NativeUI/PauseMenu/Tabs/BaseTab.cs
using System;
using System.Collections.Generic;
using System.Drawing;

namespace ScaleformUI.PauseMenu
{
    public class BaseTab
    {
        public BaseTab(string name)
        {
            Title = name;
        }

        public virtual bool Visible { get; set; }
        public virtual bool Focused { get; set; }
        public string Title { get; set; }
        public bool Active { get; set; }
        public TabView Parent { get; set; }

        public List<TabLeftItem> LeftItemList = new List<TabLeftItem>();

        public event EventHandler Activated;
        public event EventHandler DrawInstructionalButtons;

        public void OnActivated()
        {
            Activated?.Invoke(this, EventArgs.Empty);
        }
    }
}

File: ScaleformUI_Lua/src/PauseMenu/Tabs/BaseTab.lua
BaseTab = setmetatable({}, BaseTab)
BaseTab.__index = BaseTab
BaseTab.__call = function()
    return "BaseTab", "BaseTab"
end

function BaseTab.New(title, type)
    local data = {
        Title = title or "",
        Type = type or 1,
        Visible = false,
        Focused = false,
        Active = false,
        Parent = nil,
        LeftItemList = {},
        Activated = function(item) end
    }
    return setmetatable(data, BaseTab)
end